### PR TITLE
[SPARK-31315][SQL][FOLLOWUP][MINOR] Fix some typo and improve comments.

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
This is a minor PR used to fix some typo and improve comments mentioned with https://github.com/apache/spark/pull/28081/files#r402874997

### Why are the changes needed?
Fix some typo and improve comments.

### Does this PR introduce any user-facing change?
'No'.

### How was this patch tested?
Jenkins test.

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala [hunk 1]
```diff
@@ -1309,9 +1309,9 @@ object CodeGenerator extends Logging {
   // bytecode instruction
   final val MUTABLESTATEARRAY_SIZE_LIMIT = 32768
 
-  // The java source code generated by whole codegen is compiled on the Driver side and sent to
-  // each Executor as the execution code to process data. This is very effective in processing
-  // large amounts of data in a distributed environment. However, in the test environment,
+  // The Java source code generated by whole-stage codegen on the Driver side is sent to each
```
Comment (srowen) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala:1312]: So it's generated on the driver and compiled on the executor, you're sure? seems OK

Comment (beliefer) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala:1312]: Yes. Although compilation occurs on the driver too, it's just used to check.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
cc @rednaxelafx since the comment is from you.

**Comment by beliefer** [comment]
cc @cloud-fan @rednaxelafx This PR will address the issue discussed in https://github.com/apache/spark/pull/28081/files#r402874997

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala (hunk 1 above)
Comment (rednaxelafx) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala:1312]: LGTM, thank you very much!

**Review by rednaxelafx — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala (hunk 1 above)
Comment (beliefer) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala:1312]: Thanks for you remind too.

**Comment by beliefer** [comment]
@HyukjinKwon @rednaxelafx @srowen Thanks for all!

**Pull request closed without merging**